docker-pushmi-pullyu

```diff
@@ -87,14 +88,19 @@ registry_container_name=$(
   docker run --detach --publish="$registry_port:$registry_port" registry:2
 )
 
-echo "Pushing $image_name to $registry_host:$registry_port..."
-docker tag "$image_name" "$registry_host:$registry_port/$image_name"
-docker push "$registry_host:$registry_port/$image_name"
-docker rmi "$registry_host:$registry_port/$image_name"
+for image_name in $image_names; do
+  echo "Pushing $image_name to $registry_host:$registry_port..."
+  docker tag "$image_name" "$registry_host:$registry_port/$image_name"
+  docker push "$registry_host:$registry_port/$image_name"
+  docker rmi "$registry_host:$registry_port/$image_name"
+done
+
+echo "Pulling $image_names onto $deploy_target from $registry_host:$registry_port..."
 
-echo "Pulling $image_name onto $deploy_target from $registry_host:$registry_port..."
 ssh -R "$registry_port:$registry_host:$registry_port" ${ssh_opts:-} "$deploy_target" sh <<EOF
-  docker pull "$registry_host:$registry_port/$image_name" \
-    && docker tag "$registry_host:$registry_port/$image_name" "$image_name" \
-    && docker rmi "$registry_host:$registry_port/$image_name"
+  for image_name in $image_names; do
+    docker pull "$registry_host:$registry_port/$image_name" \
+      && docker tag "$registry_host:$registry_port/$image_name" "$image_name" \
+      && docker rmi "$registry_host:$registry_port/$image_name"
```

Comment:
There was a bug here. `$image_name` is substituted on your local machine and its value is the last image in the list (from the previous loop starting on line 91). This meant only the last image in the list would be pulled to the destination host.

The fix was to escape the `$` in this variable so it's substituted on the server instead, which makes this loop do the right thing. See ded4698fe104633f8adaa0a317a4b7d8ecf1761e.

Reply:
ah. nice catch.